tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -304,17 +304,31 @@ INSTANTIATE_TEST_SUITE_P(
         ::testing::Values(true),
         ::testing::Bool()));
 
+// Different scheduling modes used in
+// PipelineTestStagedReduction.StagedReduction
 enum class SchedulingMode {
+  // Manual interdevice scheduling, no intra-device scheduling
   InterDeviceOnly,
+  // Manual inter-/intra-device scheduling
   Manual,
+  // Manual inter-device scheduling, composed with ReductionScheduler
+  // intra-device schedule
+  ReductionScheduler,
```

Comment:
```suggestion
  ReductionOnly,
```

Or `Predefined` in case you'll soon apply more than just the reduction scheduler.

Reply:
No problem to change. But actually I think the word "auto" or "automatic" should be present for clarity. What about "AutoReductionSchedulingOnly"?

Reply:
Up to you. 

I don't think any of these names (ReductionScheduler, ReductionOnly, AutoReductionSchedulingOnly, or ReductionAutoScheduler) is ambiguous, so I'd optimize for conciseness and therefore proposed ReductionOnly. "Only" is to emphasize we are not testing running other schedulers at this moment. I'll be probably fine with simply `Reduction`. 

"Auto" has different meanings for different people. For me, applying just the reduction scheduler is a "manual" selection, and running FEC with segmentation and all schedulers turned on is "auto". But I'm sure you want it to be present for good reasons.

Reply:
I changed to `ReductionOnly` as you suggested